### DISCUSSION TITLE
What is the best way to invite people to my  private channel without sharing a link?

### DISCUSSION THREAD

**Discussion by rafaelfaria** [gram-js/gramjs#404] [General]
So I am creating a self-service system that would allow people to get access to my private group. I am out of ideas on how to make this bulletproof for someone to sharing the link of my group.

I had gone through a few different scenarios

**First attempt - simple approach**
I was simply sharing this group on my website but soon enough, people started sharing this outside the paid users. So I had to tweak it.

**Second attempt**
I then created a bot, where it would identify if their telegramId registered in our system and it would create a link on demand for 1 person and would last only for 30 seconds. 

I thought this would definitively stop people from sharing, but I was wrong once again.

My bot would share the link, and in theory, someone could join for the first time, then come back later and generate another link and share it to a friend or someone else during that period of 30 seconds. Android allows you to get links from buttons generated through the bot, which obviously, someone already figured this out.

**Third attempt**

I thought i could create a button that would invite the user if they click on it, but another fail, as people have option to stop being invited to channels and this wouldn't work for a lot of people (at least this is what i concluded from my tests)

**Other ideas**
1. I thought about creating a link for each one of the users, without an expiration and fixed for the person so they couldn't share, but unsure if this will scale (like for 10k users)
2. Another idea is to have a link created for 30 days and recycle if that's the case.

As you can tell, i have tried TOO many ways, with the rise of so many apps using closed groups on telegram i wish they come up with a much simpler approach, some sort of auth integration to enter a channel or something.

I would love to hear your thoughts on what worked for you, and what ideas you want to suggest. Completely out of ideas myself!

**Comment by painor** [comment]
Why don't you limit the usage of the link to only 1 user? 
![image](https://user-images.githubusercontent.com/30215770/190915593-a18567a0-aac3-41ef-ae4c-35a90bf7b289.png)

Someone pays and they get a 1 time use link. it doesn't matter if they join or if they share the link because only one person is able to join.

**Reply by rafaelfaria** [reply]
Hi @painor  as i mentioned, this was one of my ideas, but I am unsure if this scale, like for 10-20 thousand people. Do you know where I can get this information? or do you know what is the limit?

**Reply by rafaelfaria** [reply]
https://limits.tginfo.me/en I found this but i can't see anything about links generated.
